### ISSUE TITLE
bug: Getting 411 when creating s3 bucket with quarkus sdk

### ISSUE THREAD

**Issue by ayeganyan** [localstack/localstack#8203]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Getting "Length required", 411 error when creating bucket in s3 with quarkus client.
Here is the exception error: AwsErrorDetails(errorMessage=Length Required, errorCode=411, serviceName=S3)

### How are you starting LocalStack?

With a `docker run` command

### Steps To Reproduce

Start localstack with docker 
```shell
docker run --rm -p 4566:4566 -p 4510-4559:4510-4559 localstack/localstack:2.0.2
```

Here are maven configuration:
```xml
        <!-- pom.xml -->        
        <dependency>
            <groupId>io.quarkiverse.amazonservices</groupId>
            <artifactId>quarkus-amazon-s3</artifactId>
        </dependency>
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>url-connection-client</artifactId>
        </dependency>
        <dependency>
            <groupId>software.amazon.awssdk</groupId>
            <artifactId>sfn</artifactId>
            <exclusions>
                <!-- we use the `url-connection-client` so exclude the ones we don't need -->
                <exclusion>
                    <groupId>software.amazon.awssdk</groupId>
                    <artifactId>apache-client</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```
And usage:
```java
import software.amazon.awssdk.services.s3.S3Client;
import software.amazon.awssdk.services.s3.model.CreateBucketRequest;

// ..... some irrelevant code

var client = S3Client.builder()
                .endpointOverride(localstack.getEndpoint())
                .region(Region.of(localstack.getRegion()))
                .credentialsProvider(localstackCredentials(localstack))
                .build();
client.createBucket(CreateBucketRequest.builder().bucket(bucketName).build());
```

### Environment

```markdown
- OS: MacOS Ventura 13.3.1 (22E261)
- LocalStack: 2.0.2
```

**Comment by bentsku** [comment]
Hello @ayeganyan and thanks for your report!

Interestingly, we already had this issue and this should have been fixed with #8082, and that had fixed it for the Quarkus user after pulling the image with the fix.

Would it be possible to also provide the LocalStack logs when this issue happens? You should set the verbosity with `LS_LOG=trace` and `DEBUG=1`, that would be great!

Also, could you try running `curl -XPUT http://localhost:4566/testbucket` and see if that raises the same exception? Thanks!

**Comment by ayeganyan** [comment]
Hi @bentsku 
I confirm that issue fixed on localstack:2.0.2 (I did test with 2.0.0, my bad), I'll close this ticket
